spring-kafka-docs/src/main/asciidoc/kafka.adoc

```diff
@@ -1279,6 +1279,213 @@ The consumer will be paused (no new records delivered) until all the offsets for
+        };
+    }
+
+    @Bean
+    ReactiveListenerAdapter<String, String> adapter(ConcurrentKafkaListenerContainerFactory<?, ?> factory) {
+        factory.getContainerProperties().setAsyncAcks(true);
+        Random random = new Random(System.currentTimeMillis());
+        ReactiveListenerAdapter<String, String> adapter = new ReactiveListenerAdapter<>() {
+
+            @Override
+            public Mono<ConsumerRecord<String, String>> assemble(Mono<ConsumerRecord<String, String>> mono) {
+                return mono
+                        .doOnNext(rec -> {
+                            log.info("Received: {}", ListenerUtils.recordToString(rec, true));
+                            randomSleep(random);
+                            log.info("acking:   {}", ListenerUtils.recordToString(rec, true));
+                        });
+            }
+
+            private void randomSleep(Random random) {
+                try {
+                    Thread.sleep(random.nextInt(3000));
+                }
+                catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                }
+            }
+
+        };
+        adapter.setParallel(10);
+        return adapter;
+
+    }
+
+}
+
+@Component
+class Listener {
+
+    private final ReactiveListenerAdapter<String, String> adapter;
+
+    public Listener(ReactiveListenerAdapter<String, String> adapter) {
+        this.adapter = adapter;
+    }
+
+    @KafkaListener(id = "skReactor", topics = "skReactor")
+    public void listen(ConsumerRecord<String, String> record, Acknowledgment ack) {
+        this.adapter.onMessage(record, ack);
+    }
+
+}
+----
+====
+====
+[source, properties]
+----
+logging.level.org.springframework.kafka=debug
+logging.level.org.springframework.kafka.listener.adapter=off
+logging.pattern.console=[%t] %m%n
+
+spring.kafka.consumer.max-poll-records=10
+spring.kafka.consumer.auto-offset-reset=earliest
+spring.kafka.listener.ack-mode=manual
+spring.kafka.listener.poll-timeout=5s
+----
+====
+
+====
+[source, log]
+----
+[skReactor-0-C-1] skReactor: partitions assigned: [skReactor-0]
+[skReactor-0-C-1] Received: 10 records
+[skReactor-0-C-1] Commit list: {}
+[skReactor-0-C-1] Pausing for incomplete async acks: {skReactor-0=[620, 621, 622, 623, 624, 625, 626, 627, 628, 629]}
+[skReactor-0-C-1] Paused consumption from: [skReactor-0]
+[parallel-2] Received: skReactor-0@621
+[parallel-8] Received: skReactor-0@627
+[parallel-1] Received: skReactor-0@620
+[parallel-10] Received: skReactor-0@629
+[parallel-4] Received: skReactor-0@623
+[parallel-5] Received: skReactor-0@624
+[parallel-9] Received: skReactor-0@628
+[parallel-6] Received: skReactor-0@625
+[parallel-7] Received: skReactor-0@626
+[parallel-3] Received: skReactor-0@622
+[parallel-8] acking:   skReactor-0@627
+[parallel-7] acking:   skReactor-0@626
+[parallel-3] acking:   skReactor-0@622
+[parallel-1] acking:   skReactor-0@620
+[skReactor-0-C-1] Received: 0 records
```

Comment:
This is not related to the topic of this PR, but I see such a message in logs confusing: when we are paused, we definitely not going to receive any message. 
So, probably don't log it at all during pause period, or log something else indicating that we don't have data because it is paused.